### ISSUE TITLE
Can't use generated bitflags to initialise a const

### ISSUE THREAD

**Issue by PhilipDaniels** [bitflags/bitflags#180]
This works in plain Rust

```rust
const MAMMAL: u8 = 1;
const AQUATIC: u8 = 2;
const SEAL: u8 = MAMMAL | AQUATIC;
```

but the equivalent when using bitflags does not

```rust
use bitflags::bitflags;

bitflags! {
    pub struct FLAGS: u8 {
        const MAMMAL  = 1;
        const AQUATIC = 2;
    }
}

const SECOND_SEAL: FLAGS = FLAGS::MAMMAL | FLAGS::AQUATIC;
```

You get an error *error[E0015]: calls in constants are limited to constant functions, tuple structs and tuple variants* when trying to BITOR the two flags.

It would be nice if it did, because in my program I have a large initial array of flags to create and there are many different combinations making it impractical to spell them all out inside the `bitflags!` call.

Edit: Simplified the repro.

**Comment by KodrAus** [comment]
So this is a little unfortunate. It seems like a limitation in Rust's const evaluation where operators on primitive types are specially understood as being available in a constant context. We can't mark our `BitOr` implementation as being constant, even though it is.

One workaround is to do something like this:

```rust
const SECOND_SEAL: FLAGS = FLAGS::from_bits_truncate(FLAGS::MAMMAL.bits() | FLAGS::AQUATIC.bits());
```

**Comment by PhilipDaniels** [comment]
Thanks.

I wondered whether it was possible to mark `BitOr` as `const fn` or something...but I don't follow the rustc improvements that closely. I worked around my issue my not using bitflags, instead I declared an empty struct and gave it some associated constants.

**Comment by ArekPiekarz** [comment]
If it is a Rust limitation, should an issue about it be reported on their github?

**Comment by KodrAus** [comment]
There's some experimental work to support const trait impls that should enable using bitops here: https://github.com/rust-lang/rust/issues/67792 but it sounds like there's still a lot of work to be done before that will reach us.

In the meantime, we just merged in #244 which includes some nicer const set operators that avoids the need to wrap integers. So your second example could become:

```rust
const SECOND_SEAL: FLAGS = FLAGS::MAMMAL.union(FLAGS::AQUATIC);
```
